toolchain/semantics/semantics_node_block_stack.h

```diff
@@ -51,6 +60,11 @@ class SemanticsNodeBlockStack {
     return Pop();
   }
 
+  // Returns whether the current block is statically reachable.
+  auto CurrentBlockIsReachable() -> bool {
+    return stack_.empty() || stack_.back() != SemanticsNodeBlockId::Unreachable;
```

Comment:
```suggestion
    CARBON_CHECK(!stack_.empty());
    return stack_.back() != SemanticsNodeBlockId::Unreachable;
```

I believe this query should only ever be run when non-empty. i.e., this could be a CHECK if desired, although `Peek()` doesn't check first.

Reply:
Done. I moved the check into `Peek` and switched this to call `Peek`.